### ISSUE TITLE
Questions on setTimeout and "current code"

### ISSUE THREAD

**Issue by paroche** [javascript-tutorial/en.javascript.info#1282]
There are a few things that I think could be clarified:

1. In the body of the article and the summary, when it is stated that setTimeout executes "only after the current code is complete", it is always in the context of "zero delay" setTimeout. But then in to task 2 it is stated that it applies to "**any** setTimeout" (the example is for 100ms), and it looks to me like that delay is a requirement for both that task, and for the nested solution to task 1.

2. What exactly constitutes the "current code"? The next line, or the rest of the block in which the setTimeout call appears, or ... ? All the examples were for just one line of code (though in one case it's a for loop block). I could do tests, but I thought others might wonder also.

3. Does this also apply to setInterval? I know I could test this too, but might be good to mention it, one way or the other, in the article.

**Comment by iliakan** [comment]
1. This rule applies to both `setInterval` and `setTimeout`, with any delay.
2. The current code is "the currently executing script".

May I ask you to add these details to the article where you feel they are appropriate?

**Comment by paroche** [comment]
Ah, the whole script. That actually did occur to me later, especially after seeing how certain tests ran. Yeah, I could do that.

**Comment by paroche** [comment]
Comme ca?

**Comment by paroche** [comment]
Made a PR. Then realized I left out the bit with which I started this thread -- the 'current code' delay for non-zero delay cases. But I'll wait until the current PR has been processed. Shouldn't affect what I've already done.
